pkg/kubelet/dockershim/network/plugins.go

```diff
@@ -279,13 +287,16 @@ func GetPodIPs(execer utilexec.Interface, nsenterPath, netnsPath, interfaceName
 	)
 	for _, addrType := range []string{"-4", "-6"} {
 		if ip, err := getOnePodIP(execer, nsenterPath, netnsPath, interfaceName, addrType); err == nil {
-			list = append(list, ip)
+			if ip != nil {
+				list = append(list, ip)
+			}
 		} else {
+			klog.InfoS("Failed to get podIP", "netnsPath", netnsPath, "addrType", addrType, "err", err)
 			errs = append(errs, err)
 		}
 	}
 
-	if len(list) == 0 {
+	if len(errs) != 0 {
```

Comment:
Does it also need to be considered if errs length is 0 and the list length is 0?

Reply:
for function GetPodNetworkStatus，if len(ips) == 0 ，it will return error.
so for function GetPodIPs，it‘s unnecessary to check if the list length is 0.


// TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
// Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
func (plugin *cniNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
	netnsPath, err := plugin.host.GetNetNS(id.ID)
	if err != nil {
		return nil, fmt.Errorf("CNI failed to retrieve network namespace path: %v", err)
	}
	if netnsPath == "" {
		return nil, fmt.Errorf("cannot find the network namespace, skipping pod network status for container %q", id)
	}

	ips, err := network.GetPodIPs(plugin.execer, plugin.nsenterPath, netnsPath, network.DefaultInterfaceName)
	if err != nil {
		return nil, err
	}

	if len(ips) == 0 {
		return nil, fmt.Errorf("cannot find pod IPs in the network namespace, skipping pod network status for container %q", id)
	}

	return &network.PodNetworkStatus{
		IP:  ips[0],
		IPs: ips,
	}, nil
}